src/basic/stdio-util.h

```diff
@@ -9,8 +9,12 @@
 #include "macro.h"
 #include "memory-util.h"
 
-#define snprintf_ok(buf, len, fmt, ...) \
-        ((size_t) snprintf(buf, len, fmt, __VA_ARGS__) < (len))
+#define snprintf_ok(buf, len, fmt, ...)                                \
+        ({                                                             \
+                size_t _len = (len);                                   \
+                int _snpf = snprintf((buf), _len, (fmt), __VA_ARGS__); \
+                _snpf >= 0 && (size_t) _snpf < _len ? buf : NULL;      \
```

Comment:
If you care about len getting evaluated twice, shouldn't you also for buf? Or at least use (buf) here?

Reply:
indeed